utils/src/main/kotlin/Utils.kt

```diff
@@ -567,3 +571,51 @@ fun Throwable.showStackTrace() {
     // https://discuss.kotlinlang.org/t/if-operator-in-function-expression/7227.
     if (printStackTrace) printStackTrace()
 }
+
+
+/**
+ * Return Closable which temporarily moves the given directories to a temporary directory on initialization. On close
+ * it move them back to their original location overwriting any conflicting existing directory.
+ */
+fun stashDirectories(vararg directories: File) : Closeable {
+    val directorySet = HashSet(Arrays.asList(*directories))
+    return StashDirectory(directorySet)
+}
+
+private class StashDirectory(directories: Set<File>) : Closeable {
+    private val stashedDirectories = mutableMapOf<File, File>()
+
+    init {
+        directories.forEach() {
+            stashDir(it)
+        }
+    }
+
+    override fun close() {
+        HashMap(stashedDirectories).forEach() {
```

Comment:
Why wrapping this in a `HashMap`?

Reply:
Also, again omit `()` here.

Reply:
Wrapped in a HashMap in order to avoid concurrent modification exception
as for each is based on iterators which become invalidated by the modification.